sklearn/metrics/tests/test_pairwise.py

```diff
@@ -1038,12 +1049,18 @@ def test_gower_distances():
 
     D = gower_distances(X, categorical_features=[True, True])
 
-    n_rows, n_cols = np.shape(X)
-    D_expected = np.zeros((n_rows, n_rows))
-    for i in range(0, n_rows):
-        for j in range(0, n_rows):
-            D_expected[i][j] = ([1, 0][X[i][0] == X[j][0]] +
-                                [1, 0][X[i][1] == X[j][1]]) / n_cols
+    D_expected = np.zeros((4, 4))
+    # This represents the number of non missing cols for each X, Y line
+    non_missing_cols = [2, 2, 2, 0]
+    for i in range(0, 4):
+        for j in range(0, 4):
+            sum = ([1, 0][X[i][0] == X[j][0]] +
+                   [1, 0][X[i][1] == X[j][1]])
+
+            D_expected[i][j] = np.divide(sum, non_missing_cols[j],
+                                         out=np.array([np.nan]),
+                                         where=(non_missing_cols[j] != 0)
```

Comment:
Please avoid these tricky `where` params in tests. They make it harder for the reader to be sure that the test case is correct.

Reply:
Done